### DISCUSSION TITLE
Why is git blame output not coloured when piped to delta?

### DISCUSSION THREAD

**Discussion by tg0h** [dandavison/delta#1230] [Q&A] [answered]
<img width="927" alt="image" src="https://user-images.githubusercontent.com/7954426/200239394-f15c5669-f6c0-4d1e-8f81-90d49de22131.png">

`git blame <file>` produces coloured output. However, when I pipe this to delta, `git blame <file> | delta` does not provide coloured output...

I am not able to reproduce this behaviour with `git diff`.

Am I supposed to do something with the --blame-code-style and related --blame options?

**Comment by dandavison** [comment, marked as answer]
Hi @tg0h, delta uses the filename to infer the language. (E.g. `foo.py` => Python, `bah.rs` => Rust). But the output of `git blame` does not contain the filename.

However, the situation might not be hopeless. Delta does have [code that finds the command that was used when piping to delta](https://github.com/dandavison/delta/blob/master/src/utils/process.rs#L11-L19). Do you have any interest in writing Rust and would you like to have a go at making it work? I'm going to turn your question into an Issue. Thanks for documenting it so well.
